core/src/impl/Kokkos_ViewTracker.hpp

```diff
@@ -91,6 +91,9 @@ struct ViewTracker {
 
   template <class RT, class... RP>
   KOKKOS_INLINE_FUNCTION void assign(const View<RT, RP...>& vt) noexcept {
+    if (reinterpret_cast<void*>(this) ==
+        reinterpret_cast<const void*>(&vt.m_track))
```

Comment:
Why is it not a cast to ViewTracker?

Reply:
I guess we could cast to const ViewTracker*, in my mind this was clearer that its a pure numerical comparison, but my mind may be weird.

Reply:
Please change